bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1299,8 +1299,8 @@ void PreProcessor::msgProcessingLoop() {
         }
         default:
           LOG_ERROR(logger(), "Unknown message" << KVLOG(msg->type()));
-          delete msg;
       }
+      delete msg;
```

Comment:
nit:
I suggest to add comments at the end of long scopes. I'm not sure, but that's what have probably caused rthw mistake?
So here you can probably prevent future mistakes .
e.g add 
`} // switch`
and add
`} // while`